network-store-iidm-impl/src/test/java/com/powsybl/network/store/iidm/impl/RemoveDanglingSwitchesTopologyTest.java

```diff
@@ -0,0 +1,124 @@
+
+    private final List<String> removedObjects = new ArrayList<>();
+
+    @After
+    public void tearDown() {
+        removedObjects.clear();
+    }
+
+    private void addListener(Network network) {
+        network.addListener(new DefaultNetworkListener() {
+            @Override
+            public void afterRemoval(String id) {
+                removedObjects.add(id);
+            }
+        });
+    }
+
+    @Test
+    public void testRemoveAndClean() {
+        Network network = FourSubstationsNodeBreakerFactory.create();
+        addListener(network);
+        Load ld1 = network.getLoad("LD1");
+        ld1.remove(true);
+        assertEquals(List.of("LD1", "S1VL1_LD1_BREAKER", "S1VL1_BBS_LD1_DISCONNECTOR"), removedObjects);
+    }
+
+    /**
+     *   G(2)    L(1)
+     *   |       |
+     *   B1      B2
+     *   |       |
+     *   ----3----
+     *       |
+     *       D
+     *       |
+     *       BBS(0)
+     */
+    private static Network createNetworkWithForkFeeder() {
+        Network network = Network.create("test", "test");
+        VoltageLevel vl = network.newVoltageLevel()
+                .setId("VL")
+                .setNominalV(400.0)
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .add();
+        vl.getNodeBreakerView().newBusbarSection()
+                .setId("BBS")
+                .setNode(0)
+                .add();
+        vl.newLoad()
+                .setId("LD")
+                .setNode(1)
+                .setP0(0)
+                .setQ0(0)
+                .add();
+        vl.newGenerator()
+                .setId("G")
+                .setNode(2)
+                .setTargetP(0)
+                .setVoltageRegulatorOn(true)
+                .setTargetV(400)
+                .setMinP(0)
+                .setMaxP(10)
+                .add();
+        vl.getNodeBreakerView().newDisconnector()
+                .setId("D")
+                .setNode1(0)
+                .setNode2(3)
+                .add();
+        vl.getNodeBreakerView().newBreaker()
+                .setId("B1")
+                .setNode1(1)
+                .setNode2(3)
+                .add();
+        vl.getNodeBreakerView().newBreaker()
+                .setId("B2")
+                .setNode1(2)
+                .setNode2(3)
+                .add();
+        return network;
+    }
+
+    @Test
+    public void testRemoveAndCleanWithForkFeeder() {
+        Network network = createNetworkWithForkFeeder();
+        addListener(network);
+        Load ld = network.getLoad("LD");
+        ld.remove(true);
+        assertEquals(List.of("LD", "B1"), removedObjects);
+        assertNull(network.getLoad("LD"));
+        assertNull(network.getSwitch("B1"));
+        assertNotNull(network.getGenerator("G"));
+        assertNotNull(network.getBusbarSection("BBS"));
+        assertNotNull(network.getSwitch("B2"));
+        assertNotNull(network.getSwitch("D"));
+    }
+}
```

Comment:
Perhaphs a test without removeDanglingSwitches ?
And test if you remove the generatot with removeDanglingSwitches disconnector is removed ?

Reply:
This test is copied/pasted from powsybl core iidm tck, because of equipment order difference. I will remove it in the next release thanks to this PR: https://github.com/powsybl/powsybl-core/pull/1953
This is already a test in the same class of the tck, with the option to false, but if we want to improve testing we should do it in the TCK, not in this PR.